verilog/tools/kythe/scope_resolver.h

```diff
@@ -151,12 +155,19 @@ class VerticalScopeResolver : public verible::AutoPopStack<Scope*> {
 //    }
 // }
 // this way definitions can be found using the signatures.
-class FlattenedScopeResolver {
+class ScopeResolver {
  public:
-  // Searches for a VName with the given name in the scope with the given
-  // signature.
-  const VName* SearchForVNameInScope(const Signature& signature,
-                                     absl::string_view name) const;
+  explicit ScopeResolver(const ScopeResolver* previous_file_scope_resolver)
```

Comment:
// Is this allowed to be nullptr? if yes, what does that mean?  if not `ABSL_DIE_IF_NULL`, or consider taking the address of a `const&` argument as a signal to the reader.

Reply:
This is intended to be nulltr in case it is the first files which has no dependencies on other files.